kolibri/plugins/document_pdf_render/assets/src/views/pdfPage.vue

```diff
@@ -43,27 +43,43 @@
         type: Number,
         required: true,
       },
-      defaultHeight: {
+      firstPageHeight: {
         type: Number,
         required: true,
       },
-      defaultWidth: {
+      firstPageWidth: {
         type: Number,
         required: true,
       },
     },
     data: () => ({
-      height: null,
-      width: null,
       canvas: null,
       rendered: false,
     }),
     computed: {
-      pageHeight() {
-        return this.height || this.defaultHeight;
+      actualHeight() {
+        if (!this.pageReady) {
+          return null;
+        }
+        return this.pdfPage.view[3];
+      },
+      actualWidth() {
+        if (!this.pageReady) {
+          return null;
+        }
+        return this.pdfPage.view[2];
+      },
+      heightToWidthRatio() {
+        return this.actualHeight / this.actualWidth || this.firstPageHeight / this.firstPageWidth;
```

Comment:
To check I am understanding it correctly - what we are doing here is differentially scaling pages if they are of a different size, so that they all take up the same amount of vertical space, regardless of their dimensions, right?

Reply:
Exactly.

Reply:
maybe add an explanatory comment to that effect?

Reply:
whoops we're already merged